internal/pkg/manifest/transform.go

```diff
@@ -208,95 +436,72 @@ func transformPBasic() func(dst, src reflect.Value) error {
 	}
 }
 
-func transformBuildArgsOrString() func(dst, src reflect.Value) error {
-	return func(dst, src reflect.Value) error {
-		// Perform default merge
-		dstBuildArgsOrString := dst.Interface().(BuildArgsOrString)
-		srcBuildArgsOrString := src.Interface().(BuildArgsOrString)
-
-		err := mergo.Merge(&dstBuildArgsOrString, srcBuildArgsOrString, mergo.WithOverride, mergo.WithTransformers(basicTransformer{}))
-		if err != nil {
-			return err
-		}
-		dst.Set(reflect.ValueOf(dstBuildArgsOrString))
+type exclusiveTransformer struct {
+	merge                func(dst, src reflect.Value) error
+	resetExclusiveFields func(dst, src reflect.Value) error
+}
 
-		// Perform customized merge
-		dstString := dst.FieldByName("BuildString")
-		dstArgs := dst.FieldByName("BuildArgs")
+func (t exclusiveTransformer) transform(dst, src reflect.Value) error {
+	// Perform default merge
+	if err := t.merge(dst, src); err != nil {
+		return err
+	}
+	// Set or unset exclusive fields
+	if err := t.resetExclusiveFields(dst, src); err != nil {
+		return err
+	}
+	return nil
+}
 
-		srcString := src.FieldByName("BuildString")
-		srcArgs := src.FieldByName("BuildArgs")
+type basicTransformer struct{}
 
-		//` `srcArgs.IsZero()` and `srcString.IsZero()` shouldn't return true at the same time if the manifest is not malformed.
-		if !srcArgs.IsZero() {
-			dstString.Set(srcString)
-		} else if !srcString.IsNil() {
-			dstArgs.Set(srcArgs)
-		}
+// Transformer returns custom merge logic for volume's fields.
+func (t basicTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
+	return transformBasic(typ)
+}
 
-		return nil
+func opts(transformers mergo.Transformers) []func(*mergo.Config) {
+	return []func(*mergo.Config){
+		mergo.WithOverride,
+		mergo.WithTransformers(transformers),
 	}
 }
 
-func transformStringSliceOrString(originalType reflect.Type) func(dst, src reflect.Value) error {
-	return func(dst, src reflect.Value) error {
-		// Perform default merge
-		dstStruct := dst.Convert(reflect.TypeOf(stringSliceOrString{})).Interface().(stringSliceOrString)
-		srcStruct := src.Convert(reflect.TypeOf(stringSliceOrString{})).Interface().(stringSliceOrString)
-
-		err := mergo.Merge(&dstStruct, srcStruct, mergo.WithOverride, mergo.WithTransformers(basicTransformer{}))
-		if err != nil {
-			return err
+func resetExclusiveFields(dst, src reflect.Value, fieldsA, fieldsB []string) error {
```

Comment:
I feel this binary exclusiveness check might not be easily extensible when we have, for example, three elements that are mutually exclusive with each other. Can we make it more extensible like
```suggestion
func resetExclusiveFields(dst, src reflect.Value, fields ...[]string) error {
```
With this signature change, we can reduce duplicity as well.